api/authFunctions.ts

```diff
@@ -0,0 +1,25 @@
+import {account } from "./appwrite"
+import { useRouter } from 'next/navigation';
+
+   export const logout = async () => {
+    try {
+      await account.deleteSession('current');
+      const router = useRouter()
```

Comment:
You're using router twice here. It should just be a global-scoped state placed outside the function and called once.